mlflow/store/_unity_catalog/registry/uc_oss_rest_store.py

```diff
@@ -0,0 +1,371 @@
+    GenerateTemporaryModelVersionCredential,
+    GetModelVersion,
+    GetRegisteredModel,
+    ModelVersionInfo,
+    RegisteredModelInfo,
+    TemporaryCredentials,
+    UpdateModelVersion,
+    UpdateRegisteredModel,
+)
+from mlflow.protos.unity_catalog_oss_service_pb2 import UnityCatalogService
+from mlflow.store.artifact.local_artifact_repo import LocalArtifactRepository
+from mlflow.store.model_registry.base_rest_store import BaseRestStore
+from mlflow.utils._unity_catalog_oss_utils import (
+    model_version_from_uc_oss_proto,
+    registered_model_from_uc_oss_proto,
+)
+from mlflow.utils._unity_catalog_utils import (
+    get_artifact_repo_from_storage_info,
+    get_full_name_from_sc,
+)
+from mlflow.utils.annotations import experimental
+from mlflow.utils.oss_utils import get_oss_host_creds
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import (
+    _UC_OSS_REST_API_PATH_PREFIX,
+    call_endpoint,
+    extract_all_api_info_for_service,
+    extract_api_info_for_service,
+)
+from mlflow.utils.uri import is_file_uri, is_fuse_or_uc_volumes_uri
+
+_METHOD_TO_INFO = extract_api_info_for_service(UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX)
+_METHOD_TO_ALL_INFO = extract_all_api_info_for_service(
+    UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX
+)
+
+
+@experimental
+class UnityCatalogOssStore(BaseRestStore):
+    """
+    Client for an Open Source Unity Catalog Server accessed via REST API calls.
+    """
+
+    def __init__(self, store_uri):
+        super().__init__(get_host_creds=functools.partial(get_oss_host_creds, store_uri))
+        self.tracking_uri = None  # OSS has no tracking URI
+
+    def _get_response_from_method(self, method):
+        method_to_response = {
+            CreateRegisteredModel: RegisteredModelInfo,
+            CreateModelVersion: ModelVersionInfo,
+            UpdateRegisteredModel: RegisteredModelInfo,
+            DeleteRegisteredModel: DeleteRegisteredModel,
+            DeleteModelVersion: DeleteModelVersion.Response,
+            GetRegisteredModel: RegisteredModelInfo,
+            GetModelVersion: ModelVersionInfo,
+            FinalizeModelVersion: ModelVersionInfo,
+            UpdateModelVersion: ModelVersionInfo,
+            GenerateTemporaryModelVersionCredential: TemporaryCredentials,
+        }
+        return method_to_response[method]()
+
+    def _get_endpoint_from_method(self, method):
+        return _METHOD_TO_INFO[method]
+
+    def _get_all_endpoints_from_method(self, method):
+        return _METHOD_TO_ALL_INFO[method]
+
+    def create_registered_model(self, name, tags=None, description=None):
+        full_name = get_full_name_from_sc(name, None)
+        [catalog_name, schema_name, model_name] = full_name.split(".")
+        comment = description if description else ""
+        # RegisteredModelInfo is inlined in the request and the response.
+        # https://docs.databricks.com/api/workspace/registeredmodels/create
```

Comment:
Add a TODO to update this reference to UC OSS documentation when its available.

Reply:
Ping on this.

Reply:
done now